src/utils/config_filters.cpp

```diff
@@ -46,40 +46,75 @@ bool utils::config_filters::unsigned_matches_if_present(const config& filter, co
 	if(filter[attribute].empty()) {
 		return true;
 	}
+	if(cfg[attribute].empty()) {
+		return false;
+	}
 
 	return in_ranges<int>(cfg[attribute].to_int(0), utils::parse_ranges_unsigned(filter[attribute].str()));
 }
 
-bool utils::config_filters::int_matches_if_present(const config& filter, const config& cfg, const std::string& attribute)
+bool utils::config_filters::int_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool no_default_value, int def)
 {
 	if(filter[attribute].empty()) {
 		return true;
 	}
+	if(cfg[attribute].empty() && no_default_value) {
+		return false;
+	}
 
-	return in_ranges<int>(cfg[attribute].to_int(0), utils::parse_ranges_int(filter[attribute].str()));
+	return in_ranges<int>(cfg[attribute].to_int(def), utils::parse_ranges_int(filter[attribute].str()));
 }
 
 bool utils::config_filters::int_matches_if_present_or_negative(
-	const config& filter, const config& cfg, const std::string& attribute, const std::string& opposite)
+	const config& filter, const config& cfg, const std::string& attribute, const std::string& opposite, bool no_default_value, int def)
 {
-	if(int_matches_if_present(filter, cfg, attribute)) {
+	if(int_matches_if_present(filter, cfg, attribute, no_default_value , def)) {
 		return true;
 	}
 
 	// don't check for !cfg[opposite].empty(), as this assumes a default value of zero (similarly to
 	// int_matches_if_present)
```

Comment:
This documentation is outdated. However, it should be replaced by a description of the new logic, not just removed.

Reply:
updated documentation

Reply:
fixed

Reply:
Looking further at this, I think the default for this function can only be zero. The purpose of the `..._or_negative` function is to handle two opposites that are symmetrical around zero, that `add=-2` is equivalent to `sub=2`.

Reply:
but if i put a default value, then if i check with add=-1-3 an ability with value=15, then with add=0 by default it match ability then what it should'nt.